bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/AbstractTableViewer.java

```diff
@@ -1057,6 +1166,73 @@ protected void assertContentProviderType(IContentProvider provider) {
 				|| provider instanceof ILazyContentProvider);
 	}
 
+	private void createExpandableNode(Object element, int index) {
+		// case2 : if ExpNode absent. Create an ExpNode
+		Item[] items = doGetItems();
+		// there is an extra element being added.
+		Object[] allEles = new Object[items.length + 1];
+
+		// case2.1 : new element goes inside ExpNode and a new ExpNode is created
+		if (index >= doGetItemCount()) {
+			for (int i = 0; i < items.length; i++) {
+				allEles[i] = items[i].getData();
+			}
+			allEles[allEles.length - 1] = element;
+			IExpandableNode expNode = new ExpandableNode(allEles, items.length, getItemsLimit(), this);
+			createItem(expNode, -1);
+		} else {
+			// case2.2 : insert the element at it's appropriate position. update all
+			// elements until last element and create an ExpNode.
+			int j = 0;
+			for (int i = 0; i < items.length; i++) {
+				if (i == index) {
+					allEles[j = j + 1] = items[i].getData();
```

Comment:
`j = j + 1` followed by `j++` looks weird. What does this code do?

Reply:
skip the jth position in result node. that is [index] is left blank to copy later. I think i can simplify it. i will do it.